vignettes/readr.Rmd

```diff
@@ -160,61 +159,46 @@ readr comes with five parsers for rectangular file formats:
 * `read_fwf()` for fixed-width files
 * `read_log()` for web log files
 
-Each of these functions firsts calls `spec_xxx()` (as described above), and then parses the file according to that column specification:
+If readr finds problems with parsing your data, it will emit a warning message.
+For example, in our `challenge.csv` file, the column type we expected from our data was a column of dates and a column of doubles.
 
 ```{r}
 df1 <- read_csv(readr_example("challenge.csv"))
 ```
 
-The rectangular parsing functions almost always succeed; they'll only fail if the format is severely messed up. Instead, readr will generate a data frame of problems. The first few will be printed out, and you can access them all with `problems()`:
+The data was successfully imported, but we get a warning message about parsing issues.
+To view more details about these parsing issues, call `problems()` on your data frame.
+This will generate a data frame with information about where the problem was found, and why.
 
 ```{r}
 problems(df1)
 ```
 
-You've already seen one way of handling bad guesses: increasing the number of rows used to guess the type of each column. 
-
-```{r}
-df2 <- read_csv(readr_example("challenge.csv"), guess_max = 1001)
-```
-
-Another approach is to manually supply the column specification.
-
-### Overriding the defaults
-
-In the previous examples, you may have noticed that readr printed the column specification that it used to parse the file:
+In this example, one of the dates is in an incorrect format.
+Since `problems()` gives us the row number from the original file, we can view that data in R with a combination of `readr::read_lines()` and `writeLines()`.
 
 ```{r}
-#> Parsed with column specification:
-#> cols(
-#>   x = col_integer(),
-#>   y = col_character()
-#> )
+writeLines(read_lines(readr_example("challenge.csv"), skip = 1325, n_max = 3))
 ```
 
-You can also access it after the fact using `spec()`:
+We can work around this parsing error by importing the date column as character and converting it to a column of dates, after the fact. The lubridate package is a great option that we show here. But since readr doesn't depend on lubridate, we won't execute this code chunk.
```

Comment:
You actually could make this a static code chunk. That is, set it to `eval = FALSE`, but inline the current result. It's not a perfect solution, but the pros would seem to outweigh the cons.

I'm talking about showing folks this:

``` r
> df1[1325:1327, ]
# A tibble: 3 × 2
  date       value
  <date>     <dbl>
1 2008-03-16   762
2 2008-02-17   206
3 2019-01-28  1333
```

Reply:
I reiterate this:

> I think you should add some brief comment ... that it's indicative of a general type of workaround, where you import as character and use a more specialized package (or custom function) to parse a tricky column.